include/robot_interfaces/sensors/pylon_driver.hpp

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file
+ * @brief Driver to interface with the camera using Pylon and OpenCV.
+ * @copyright 2020, New York University, Max Planck Gesellschaft. All rights
+ *            reserved.
+ * @license BSD 3-clause
+ */
+
+#pragma once
+
+#include <ctime>
+#include <iostream>
+
+#include <real_time_tools/timer.hpp>
+
+#include <robot_interfaces/sensors/camera_observation.hpp>
+#include <robot_interfaces/sensors/sensor_driver.hpp>
+
+#include <pylon/PylonIncludes.h>
+#include <opencv2/opencv.hpp>
+
+namespace robot_interfaces
+{
+/**
+ * @brief Driver for interacting with a camera via Pylon and store
+ * images using OpenCV.
+ */
+class PylonDriver : public SensorDriver<CameraObservation>
+{
+public:
+    Pylon::PylonAutoInitTerm auto_init_term_;
```

Comment:
I was curious about what this thing does.  From what I understood, it takes care of initializing and terminating, so it should not be necessary to explicitly call `PylonInitialize` and `PylonTerminate` below.

Reply:
Yeah, it takes care of initializing and terminating. I added the PylonTerminate explicitly to terminate in case of runtime_error exceptions like when there is no device connected, or when the grab cannot succeed. So, added PylonInitialize to go with it. Should I remove them then?

Reply:
If the explicit terminate before throwing is needed, then go for that method.  I think in this case you can simply remove the "Auto" thing, because from what I saw you already call Initialize/Terminate everywhere properly.

Reply:
Hmm yeah, I tried doing this now, but removing the auto init results in a runtime error of "Caught an unknown exception!". I am not sure what this auto is really doing

Reply:
Okay, then just leave it there.  I guess as long as everything works, that's good enough for us :)

Reply:
ok :)